### PR TITLE
SAMZA-1279: Do not process onProcessorChange if list of children is null.

### PR DESCRIPTION
**Pull request by sborya**

### CODE REVIEW

File: samza-core/src/main/java/org/apache/samza/zk/ZkControllerImpl.java [hunk 1]
```diff
@@ -94,6 +94,12 @@ public void subscribeToProcessorChange() {
      */
     @Override
     public void handleChildChange(String parentPath, List<String> currentChildren) throws Exception {
+      if (currentChildren == null) {
+        // this may happen only in case of some exception in ZK. Exception should be visible in the logs.
```
Comment (prateekm) [inline samza-core/src/main/java/org/apache/samza/zk/ZkControllerImpl.java]: Update comment to describe when this can happen?

Comment (navina) [inline samza-core/src/main/java/org/apache/samza/zk/ZkControllerImpl.java]: can you rephrase to say- **if the parentPath doesn't exist, ZkClient doesn't throw exception. It will instead pass a "null" list.** instead of *some exception in ZK*
